File: go/internal/dtos/meeting_room.go
package dtos

type MeetingRoom struct {
	Id        string `json:"id"`
	Name      string `json:"name"`
	CreatedAt string `json:"created_at"`
}

type TimeSlot struct {
	Id        string `json:"id"`
	StartDate string `json:"start_date"`
	EndDate   string `json:"end_date"`
	CreatedAt string `json:"created_at"`
}

type MeetingRoomWithTimeSlotsResponse struct {
	Id        string     `json:"id"`
	Name      string     `json:"name"`
	CreatedAt string     `json:"created_at"`
	TimeSlots []TimeSlot `json:"time_slots"`
}

File: react/src/dtos/meetingRoom.ts
interface TimeSlot {
  id: string;
  start_date: string;
  end_date: string;
  created_at: string;
}

interface MeetingRoomWithTimeSlotsResponse {
  id: string;
  name: string;
  created_at: string;
  time_slots: TimeSlot[];
}

export type { MeetingRoomWithTimeSlotsResponse, TimeSlot };
